src/index.jst

```diff
@@ -5,7 +5,10 @@
 module.exports = function equal(a, b) {
   if (a === b) return true;
 
-  if (a && b && typeof a == 'object' && typeof b == 'object') {
+  var aType = typeof a;
+  if (aType !== typeof b) return false;
```

Comment:
I do not think type comparison is necessary.
The first strict equality will only succeed if the type is the same.
If it is not the same, a and b can only be equal if they ar both objects or they are both NaN (check in the bottom).

About the size, there is another PR that seems to be better addressing Map/Set equality than the current code, so size will no longer be used.

Going to close it.